api_types.ts

```diff
@@ -805,6 +807,65 @@ export interface Logger {
   error(message: string, ...args: LoggerParamType[]): void;
 }
 
+export enum InvocationErrorType {
+  Timeout = 'Timeout',
+
+  ResponseTooLarge = 'ResponseTooLarge',
+
+  HttpStatusError = 'HttpStatusError',
+
+  /**
+   * Could mean 3rd party API rate limit or a rate limit imposed by Coda.
+   */
+  RateLimitExceeded = 'RateLimitExceeded',
+
+  Unknown = 'Unknown',
+}
+
+interface BaseInvocationError {
+  type: InvocationErrorType;
+
+  message?: string;
+
+  errorData?: string | object;
```

Comment:
I'd like to narrow this typing soon, but the current state of our errors isn't very clean and I doubt it's worth the time right now to figure out coercing for all the permutations.